src/creds/creds.c

```diff
@@ -1220,6 +1232,71 @@ static int settle_scope(
         return 0;
 }
 
+static bool normalize_separator(char c) {
+        if (IN_SET(c, '-', '+', '_'))
+                return '_';
+
+        return c;
+}
+
+static bool enum_name_equal(const char *x, const char *y) {
+        if (x == y)
+                return true;
+        if (!x || !y)
+                return false;
+
+        for (;; x++, y++) {
+                char a = normalize_separator(*x), b = normalize_separator(*y);
+                if (a != b)
+                        return false;
+                if (a == 0)
+                        return true;
+        }
+}
+
+static CredKeyType cred_key_type_from_string_harder(const char *s) {
```

Comment:
unrelated to this, but it would be great to comment down that CRED_KEY_TYPE_ABSENT is just an alias of _NULL in enum definition.